lib/src/main/java/io/ably/lib/types/Summary.java

```diff
@@ -0,0 +1,145 @@
+package io.ably.lib.types;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import io.ably.lib.util.Log;
+import io.ably.lib.util.Serialisation;
+import org.msgpack.core.MessagePacker;
+import org.msgpack.core.MessageUnpacker;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A summary of all the annotations that have been made to the message. Will always be
+ * populated for a message.summary, and may be populated for any other type (in
+ * particular a message retrieved from REST history will have its latest summary
+ * included).
+ * The keys of the map are the annotation types. The exact structure of the value of
+ * each key depends on the aggregation part of the annotation type, e.g. for a type of
+ * reaction:distinct.v1, the value will be a DistinctValues object. New aggregation
+ * methods might be added serverside, hence the 'unknown' part of the sum type.
+ */
+public class Summary {
```

Comment:
Also, instead of utility methods, the following pattern can be followed. [Updated spec comment](https://github.com/ably/specification/pull/292#discussion_r2138110046) for the same.